packages/connection-manager/src/js/Connection/XHRConnection.js

```diff
@@ -0,0 +1,198 @@
+import AbstractConnection from "./AbstractConnection";
+import {
+  CONNECTION_EVENT_CLOSE,
+  CONNECTION_EVENT_OPEN,
+  CONNECTION_EVENT_PROGRESS,
+  CONNECTION_EVENT_ABORT,
+  CONNECTION_EVENT_ERROR,
+  CONNECTION_EVENT_SUCCESS,
+  CONNECTION_STATE_CANCELED,
+  CONNECTION_STATE_INIT,
+  CONNECTION_STATE_DONE,
+  CONNECTION_STATE_STARTED
+} from "./ConnectionConstants";
+
+export const ALLOWED_METHODS = ["GET", "POST", "PUT", "DELETE"];
+
+/**
+ * Basic XHR Connection
+ * @todo add/remove headers
+ */
+export default class XHRConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of XHRConnection
+   * @constructor
+   * @param {string} url – URL to be fetched
+   * @param {string} [_method=GET] – used method
+   * @param {mixed} [_data] – payload for request
+   * @param {string} [_header] – additional headers (like content-type)
+   */
+  constructor(url, _method = "GET", _data = null, _header = null) {
+    super(url);
+
+    var method;
```

Comment:
Please always use `const` and `let` instead of `var`... @dcos/frontend-foundation can we add a eslint rule for this one?